### PR TITLE
[Issue 2443] Complete community page in the new docusaurus website

### PR DESCRIPTION
**Pull request by Abhey**
Descriptions of the changes in this PR:

This PR introduces changes to complete the community page of the new up-coming docusaurus based website.

### Motivation

The #2426 Pull Request introduced the changes for new docuasurus based website, However few of the pages like community and project were incomplete. This PR introduces the changes to complete the community page.

### Changes

I have added the markdown files for the following web-pages - 
* bookkeeper-proposals
* coding-guide
* community-meetings
* contributing
* issue-report-guide
* mailing-lists
* papers-and-presentations
* release-guide
* release-management
* testing-guide

I have also updated the code in community.js to refactor the way tables for PMC and committers is getting generated, So that we get the best out of the react based framework.

### Testing Details - 

I have build the changes locally and have verified that all the links are rendering properly or not. Here is the screenshot of the new community web-page.

Screenshot 2020-10-19 at 18 36 34

Master Issue: #2443

### CODE REVIEW

**Comment by eolivelli** [comment]
very cool,
please remove the 'community-meetings' part, as we are no more doing them (hopefully one day we will start again)

**Review by eolivelli — CHANGES_REQUESTED** [review]
thank you.
@Ghatage  @nicoloboschi  PTAL

File: site2/docs/bookkeeper-proposals.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+[BP-30: BookKeeper Table Service](https://docs.google.com/document/d/155xAwWv5IdOitHh1NVMEwCMGgB28M3FyMiQSxEpjE-Y/edit#heading=h.56rbh52koe3f) | Accepted
+[BP-31: BookKeeper Durability Anchor](../../bps/BP-31-durability) | Accepted
+[BP-32: Advisory (optimistic) write close](../../bps/BP-32-advisory-write-close) | Accepted
+[BP-33: Move releasing docker images out of main repo](../../bps/BP-33-building-official-docker-imags) | Draft
+[BP-34: Cluster Metadata Checker](../../bps/BP-34-cluster-metadata-checker) | Accepted
+[BP-35: 128 bits support](../../bps/BP-35-128-bits-support) | Accepted
+[BP-36: Stats documentation annotation](../../bps/BP-36-stats-documentation-annotation) | Accepted
+[BP-37: Improve configuration management for better documentation](../../bps/BP-37-conf-documentation) | Accepted
+[BP-38: Publish Bookie Service Info on Metadata Service](../../bps/BP-38-bookie-endpoint-discovery) | Accepted
+[BP-41: Separate BookieId from Separate BookieId from Bookie Network Address](../../bps/BP-41-bookieid) | Draft
```
Comment (nicoloboschi) [inline site2/docs/bookkeeper-proposals.md:114]: we may add BP-42?

File: site2/website/pages/en/community.js [hunk 2]
```diff
@@ -21,277 +21,333 @@ function Help(props) {
...
+      title: "Apache BookKeeper on Slack"
     },
     {
-      content: "Learn more about our [sponsors.](http://www.apache.org/licenses/)",
-      title: 'Sponsorship',
+      content: `Learn more about the upcoming and on-going releases at [Apache BookKeeper Release Management.](${docUrl('release-management')})`,
+      title: "Release Mangement"
     },
+    {
+      content: `As a member of the Apache BookKeeper community, you are welcome to join our bi-weekly meetings. You can find the meeting schedule at [Apache BookKeeper Commnunity Meetings](${docUrl('community-meetings')})`,
```
Comment (nicoloboschi) [inline site2/website/pages/en/community.js:38]: Commnunity typo

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
I left minor comments, in overall it is okay

File: site2/docs/bookkeeper-proposals.md (hunk 1 above)
Comment (eolivelli) [reply site2/docs/bookkeeper-proposals.md:114]: @nicoloboschi IMHO it is better to focus on porting the current website.
BP page is a mess we can fix it later

**Review by Ghatage — COMMENTED** [review]
@Abhey 
Great job converting the PMC/committers to variables and dynamically rendering them.
Most of it looks good but there is one more thing I am thinking:
- While this is a direct port of the website as @eolivelli said, I feel we should put in some TODO's in comments at places at the minimum so we can follow up later once the entire website is done.
- For example, places in the website still refer to using Java 8 while we've also moved on to newer versions which have no mention.
- So I feel it would be useful to keep track of these things so we can come back to them later once the website prototype is complete.

Will take a closer look later today.

**Comment by Abhey** [comment]
@eolivelli @nicoloboschi @Ghatage Thanks for reviewing the PR.

I was fixing the broken hyperlinks and I found that many of the hyperlinks in the `bookeeper-proposals` were broken cause they were linking to the Markdown format BPs that are present in the current website directory structure.

Should I add those missing files as part of this PR and fix the hyperlinks Or should I create a new issue for adding those files?

I feel that it will be good to address this issue as a separate PR, Cause then we can also update the content of the `bookeeper-proposals`, With the instructions for where should the individuals add the markdown flavoured BP moving forward. Please do let me know your thoughts over the same.

**Comment by eolivelli** [comment]
Let's so it in a separate patch

**Comment by Ghatage** [comment]
rerun failure checks

**Comment by eolivelli** [comment]
@Ghatage please merge if you are happy with this patch.
All checks have passed.

It would be good to add an additional github action that builds the new website, this way we are sure that each patch does not break the build

cc @nicoloboschi

**Review by Ghatage — APPROVED** [review]
I just tested these out locally. They look fantastic!
Great work @Abhey! Looking forward to more of your contributions to the community!

**Comment by Abhey** [comment]
Thanks, @Ghatage for merging the PR. This my first contribution to the BookKeeper code-base, Looking forward to contribute more 🙂

**Comment by eolivelli** [comment]
Thanks to you!
We hopefully publish the new website soon with your help

**Pull request merged**